Source/WebKitLegacy/mac/Misc/WebNSDataExtras.mm

```diff
@@ -60,15 +57,16 @@ - (NSString *)_web_capitalizeRFC822HeaderFieldName
         if (replace) {
             if (!somethingChanged) {
                 somethingChanged = TRUE;
-                if (CFStringGetBytes(name, CFRangeMake(0, len), kCFStringEncodingISOLatin1, 0, FALSE, NULL, 0, NULL) == len) {
+                NSRange range = NSMakeRange(0, len);
+                if ([self getBytes:NULL maxLength:0 usedLength:nil encoding:NSISOLatin1StringEncoding options:NSStringEncodingConversionExternalRepresentation range:range remainingRange:NULL]) {
                     // Can be encoded in ISOLatin1
                     useUniCharPtr = FALSE;
-                    charPtr = static_cast<char*>(CFAllocatorAllocate(kCFAllocatorDefault, len + 1, 0));
-                    CFStringGetCString(name, charPtr, len+1, kCFStringEncodingISOLatin1);
+                    charPtr = static_cast<char *>(malloc(len + 1));
```

Comment:
I admit I don't fully understand all the changes to this function and an explanation in the changelog would be helpful (or its own PR). However, we generally use fastMalloc() / fastFree() in WebKit, not malloc()/free().

Reply:
Actually, this must be an exception; [[NSString alloc] initWithCharactersNoCopy:uniCharPtr length:len freeWhenDone:YES];

freeWhenDone calls free(); There is nothing that can be changed about this.

This has to be done via malloc/free for reasons beyond Webkit's control.

Reply:
The way it was done before also leaked memory. This portion of the PR is killing two birds with one stone.